masonry/src/render_root.rs

```diff
@@ -203,9 +203,13 @@ impl RenderRoot {
                 let last = self.last_anim.take();
                 let elapsed_ns = last.map(|t| now.duration_since(t).as_nanos()).unwrap_or(0) as u64;
                 let root_state = self.root_state();
-                if root_state.request_anim {
-                    root_state.request_anim = false;
-                    self.root_lifecycle(LifeCycle::AnimFrame(elapsed_ns));
+                if root_state.needs_anim {
```

Comment:
I wonder if this check should be inside `run_update_anim_pass`. I can understand why it isn't with `post_event_processing`, but my understanding is that that will be removed?

Reply:
I need to write down my thoughts on this (and correct the RFC); The short version is that update_anims doesn't really have the semantics of a "rewrite" pass. It's more of a special event pass.

Reply:
Re-reading your question again, I realize I misunderstood it. You're right that the check is a bit superfluous. The pass already does it once per widget anyway.